app/src/main/kotlin/cz/covid19cz/erouska/exposurenotifications/ExposureNotificationsRepository.kt

```diff
@@ -259,14 +275,46 @@ class ExposureNotificationsRepository @Inject constructor(
     }
 
     suspend fun checkExposure(context: Context) {
-        val lastExposure = getLastRiskyExposure()?.daysSinceEpoch
-        val lastNotifiedExposure = prefs.getLastNotifiedExposure()
-        if (lastExposure != null && lastExposure != lastNotifiedExposure) {
+        importLegacyExopsures()
+        db.dao().deleteOld()
+        db.dao().insert(getDailySummariesFromApi(true).map {
+            DailySummaryEntity(
+                daysSinceEpoch = it.daysSinceEpoch,
+                maximumScore = it.summaryData.maximumScore,
+                scoreSum = it.summaryData.scoreSum,
+                weightenedDurationSum = it.summaryData.weightedDurationSum,
+                importTimestamp = System.currentTimeMillis(),
+                notified = false,
+                accepted = false
+            )
+        })
+        val latestExposure = db.dao().getLatest().firstOrNull()?.daysSinceEpoch
+        val lastNotifiedExposure = db.dao().getLastNotified().firstOrNull()?.daysSinceEpoch
+        if (latestExposure != null && latestExposure != lastNotifiedExposure) {
             LocalNotificationsHelper.showRiskyExposureNotification(context)
-            prefs.setLastNotifiedExposure(lastExposure)
+            db.dao().markAsNotified()
```

Comment:
Přemýšlím - nemůže nastat nějaký edge case, že přijdou dvě exposures stejný den, nastaví se jako notified ten první a druhý den tě to notifikuje o tom druhém?

Reply:
daily summary dostaneš až po půlnoci za komplet den, takže tohle se nestane